extensions/modules/file/src/test/xquery/modules/file/sync.xqm

```diff
@@ -0,0 +1,285 @@
+ : modify it under the terms of the GNU Lesser General Public
+ : License as published by the Free Software Foundation; either
+ : version 2.1 of the License, or (at your option) any later version.
+ :
+ : This library is distributed in the hope that it will be useful,
+ : but WITHOUT ANY WARRANTY; without even the implied warranty of
+ : MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ : Lesser General Public License for more details.
+ :
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.1";
+
+module namespace sync="http://exist-db.org/xquery/test/file/sync";
+import module namespace helper="http://exist-db.org/xquery/test/util/helper" at "resource:util/helper.xqm";
+import module namespace fixtures="http://exist-db.org/xquery/test/util/fixtures" at "resource:util/fixtures.xqm";
+
+declare namespace test="http://exist-db.org/xquery/xqsuite";
+
+declare variable $sync:suite := "sync";
+
+declare
+    %test:setUp
+function sync:setup() as empty-sequence() {
+    helper:setup-db()
+};
+
+declare
+    %test:tearDown
+function sync:tear-down() {
+    helper:clear-db(),
+    helper:clear-suite-fs($sync:suite)
+};
+
+declare
+    %test:assertTrue
+function sync:simple() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options(())
+    => helper:assert-sync-result(map {
+        "updated": $fixtures:ALL-UPDATED,
+        "deleted": (),
+        "fs": $fixtures:ROOT-FS
+    })
+};
+
+declare
+    %test:assertTrue
+function sync:empty-options-map() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options(map{})
+    => helper:assert-sync-result(map {
+        "updated": $fixtures:ALL-UPDATED,
+        "deleted": (),
+        "fs": $fixtures:ROOT-FS
+    })
+};
+
+declare
+    %test:assertError
+function sync:deprecated-options() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options($fixtures:mod-date)
+    => helper:assert-sync-result(map {
+        "updated": $fixtures:ALL-UPDATED,
+        "deleted": (),
+        "fs": $fixtures:ROOT-FS
+    })
+};
+
+declare
+    %test:assertError("err:XPTY0004")
+function sync:bad-options-1() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options(xs:date("2012-12-21"))
+};
+
+declare
+    %test:assertError("err:XPTY0004")
+function sync:bad-options-2() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options("2012-12-21T10:12:21")
+};
+
+declare
+    %test:assertError("err:XPTY0004")
+function sync:bad-options-3() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options("lizard")
+};
+
+declare
+    %test:assertError("err:XPTY0004")
+function sync:bad-options-4() {
+    helper:get-test-directory($sync:suite)
+    => helper:sync-with-options("")
```

Comment:
I would suggest removing the arrow operator here, it will make Java debugging of this code almost impossible to follow - test code should be as simple as possible rather than clever.